.github/workflows/release.yaml

```diff
@@ -16,6 +16,9 @@ jobs:
   test:
     name: Build and Test
     uses: ./.github/workflows/build-and-test.yaml
```

Comment:
zizmor would probably warn here.
the referenced flow caches wheels, which means someone can run attack this:
- open a PR with very same uv.lock file
- includes bad code that pollutes the wheel house
- waits for release job to run and use crooked wheels